mongo/gridfs/bucket.go

```diff
@@ -465,23 +503,25 @@ func createIndexIfNotExists(ctx context.Context, iv mongo.IndexView, model mongo
 		_ = c.Close(ctx)
 	}()
 
+	modelKeysBytes, err := bson.Marshal(model.Keys)
+	if err != nil {
+		return err
+	}
+	modelKeysDoc := bsoncore.Document(modelKeysBytes)
+
 	var found bool
-	for c.Next(ctx) {
+	for c.Next(ctx) && !found {
```

Comment:
I see why we made this change, but would it actually be easier to `return nil` inside the loop if `indexDocsEqual` returns true and get rid of this `found` variable altogether?